torch/optim/rmsprop.py

```diff
@@ -81,7 +89,10 @@ def step(self, closure=None):
                 state['step'] += 1
 
                 if group['weight_decay'] != 0:
-                    grad = grad.add(group['weight_decay'], p.data)
+                    if group['l2_reg']:
+                        grad.add_(group['weight_decay'], p.data)
+                    else:
+                        p.data.add_(-group['weight_decay'], p.data)
```

Comment:
```suggestion
                        p.data.add_(-group['weight_decay'] * group['lr'], p.data)
```

In L2, the weight decay is stored inside the grad variable, which is later multiplied by the learning rate. This new formulation is not multiplying it by the learning rate (see Algorithms 1 and 2 in the paper).